File: 7_kyu/Cats_and_shelves.go
package kata

// Calculates number of jumps from start to finish.
func Cats(start, finish int) int {
  if finish < start {
    panic("argument error: finish < start")
  }
  delta := finish - start
  // Cat can jump over one shelf "/3" but must go the last steps "delta %3"
  return delta / 3 + delta % 3
}

File: 7_kyu/Cats_and_shelves.rs
// Calculates number of jumps from start to finish.
pub fn cats_and_shelves(start: u8, finish: u8) -> u8 {
  if finish < start {
    panic!("argument error: finish < start");
  }
  let delta = finish - start;
  // Cat can jump over one shelf "/3" but must go the last steps "delta %3"
  delta / 3 + delta % 3
}
